src/components/scenes/CryptoExchangeQuoteScene.tsx

```diff
@@ -69,27 +107,69 @@ export const CryptoExchangeQuoteScene = (props: Props) => {
     logEvent('Exchange_Shift_Quote')
     swapVerifyTerms(swapConfig)
       .then(async result => {
-        if (!result) await dispatch(exchangeTimerExpired(navigation, quote, onApprove))
+        if (!result) await dispatch(exchangeTimerExpired(navigation, selectedQuote, onApprove))
       })
       .catch(err => showError(err))
 
     return () => {
-      if (!calledApprove) quote.close().catch(err => showError(err))
+      if (!calledApprove) selectedQuote.close().catch(err => showError(err))
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
   const doShift = async () => {
     setCalledApprove(true)
-    await dispatch(shiftCryptoCurrency(navigation, quote, onApprove))
+    await dispatch(shiftCryptoCurrency(navigation, selectedQuote, onApprove))
   }
 
   const renderTimer = () => {
-    const { expirationDate } = quote
+    const { expirationDate } = selectedQuote
     if (!expirationDate) return null
-    return <CircleTimer timeExpired={async () => await dispatch(exchangeTimerExpired(navigation, quote, onApprove))} expiration={expirationDate} />
+    return <CircleTimer timeExpired={async () => await dispatch(exchangeTimerExpired(navigation, selectedQuote, onApprove))} expiration={expirationDate} />
   }
 
+  const renderRow = useHandler((item: { item: EdgeSwapQuote; section: Section; index: number }) => {
+    const quote = item.item
+    return (
+      <TouchableOpacity
+        onPress={() => {
+          navigation.replace('exchangeQuote', {
```

Comment:
Instead of a full `navigation.replace`, would have been simpler to make selectedQuote a `useState` which you then change with a `setSelectedQuote`

Line 48 could be

`const { selectedQuote: initialSelectedQuote, quotes, onApprove } = route.params`

Then 

`const [ selectedQuote, setSelectedQuote ] = useState(initialSelectedQuote)`